apps/api-extractor/src/analyzer/TypeScriptHelpers.ts

```diff
@@ -209,4 +209,37 @@ export class TypeScriptHelpers {
 
     return highest;
   }
+
+  private static readonly _wellKnownSymbolNameRegExp: RegExp = /^__@\w+$/;
+
+  /**
+   * Returns whether the provided name was generated for a built-in ECMAScript symbol.
+   */
+  public static isWellKnownSymbolName(name: string): boolean {
+    return TypeScriptHelpers._wellKnownSymbolNameRegExp.test(name);
+  }
+
+  private static readonly _uniqueSymbolNameRegExp: RegExp = /^__@.*@\d+$/;
+
+  /**
+   * Returns whether the provided name was generated for a TypeScript `unique symbol`.
+   */
+  public static isUniqueSymbolName(name: string): boolean {
+    return TypeScriptHelpers._uniqueSymbolNameRegExp.test(name);
+  }
+
+  public static tryGetLateBoundName(declarationName: ts.ComputedPropertyName): string | undefined {
+    // Only a ComputedPropertyName whose expression is an EntityNameExpression can be
+    // used as a late-bound name.
+    let expressionName: string = '';
+    let expression: ts.EntityNameExpression = declarationName.expression as ts.EntityNameExpression;
+    while (expression.kind === ts.SyntaxKind.PropertyAccessExpression) {
+      expressionName = `.${expression.name.getText().trim()}${expressionName}`;
+      expression = expression.expression;
+    }
+    if (expression.kind !== ts.SyntaxKind.Identifier) {
+      return undefined;
+    }
+    return `[${expression.getText().trim()}${expressionName}]`;
```

Comment:
Why do we use `expression.getText()` here, but `expression.name.getText()` above?

Reply:
In both cases we only want to get the text of an `Identifier`. Above we are looking at a `PropertyAccessExpression`, and we want to encode the text of the `.name` property (which is an `Identifier`). At this line, `expression` is an `Identifier` so we get its own text. This is to avoid including any possible comments that could have been added inside of the computed property name.

That said, I'm actually going to replace this loop with a call to the TypeScript node printer so that it can handle the logic instead.